ignite/engine/__init__.py

```diff
@@ -4,12 +4,13 @@
 from ignite._utils import convert_tensor
 
 
-def _prepare_batch(batch, device=None):
+def _prepare_batch(batch, device=None, non_blocking=False):
     x, y = batch
-    return convert_tensor(x, device=device), convert_tensor(y, device=device)
+    return (convert_tensor(x, device=device, non_blocking=non_blocking),
+            convert_tensor(y, device=device, non_blocking=non_blocking))
 
 
-def create_supervised_trainer(model, optimizer, loss_fn, device=None):
+def create_supervised_trainer(model, optimizer, loss_fn, device=None, non_blocking=False):
```

Comment:
Can we add `non_blocking` to the docstring

Reply:
Incorporated and committed.